### PR TITLE
Fix some account_tx issues: (RIPD-1035)

### PR DESCRIPTION
**Pull request by JoelKatz**
* Sanely handled specified ledger in account_tx
* Reject un-validated ledger in account_tx
* Wait to publish a ledger until it's indexed
* Add unit test for PendingSaves

### CODE REVIEW

File: src/ripple/app/ledger/PendingSaves.h [hunk 1]
```diff
@@ -36,49 +37,117 @@ class PendingSaves
...
-    getSnapshot() const ->
-        decltype(set_)
+    /** Get a snapshot of the pending saves
+
+        Each entry in the returned map corresponds to a ledger
+        that is in progress or dispatched. The boolean indicates
+        whether work is currently in progress.
+    */
+    std::map <LedgerIndex, bool>
+    getSnapshot () const
```
Comment (vinniefalco) [inline src/ripple/app/ledger/PendingSaves.h]: Unused?

Comment (JoelKatz) [inline src/ripple/app/ledger/PendingSaves.h]: It's called in LedgerMaster.cpp, around line 475 or so.

**Comment by JoelKatz** [comment]
Aww, I was so excited about seeing a microscopic increase in code coverage but it seems a spurious failure is going to deny me this small pleasure.

**Comment by vinniefalco** [comment]
Missing `#include <beast/unit_test/suite.h>`

File: src/ripple/app/ledger/PendingSaves.h [hunk 2]
```diff
@@ -36,49 +37,117 @@ class PendingSaves
...
+        await_.notify_all();
+    }
+
+    /** Check if a ledger is in progress
+
+        This reports whether a ledger is in progress
+        or not.
+
+        @return 'true' if work is in progress or pending
+    */
```
Comment (vinniefalco) [inline src/ripple/app/ledger/PendingSaves.h]: This entire comment could be replaced with one line:
```
/** Return `true` if a ledger is in the progress of being saved. */
```

**Comment by nbougalis** [comment]
Merged as 5ee94f89281091b46ee15eb16a604aba5dd318a2 (with Vinnie's comment rewording incorporated).

**Pull request closed without merging**